llvm/lib/Target/AMDGPU/AMDGPUISelLowering.cpp

```diff
@@ -3062,6 +3068,25 @@ static bool isCttzOpc(unsigned Opc) {
   return Opc == ISD::CTTZ || Opc == ISD::CTTZ_ZERO_UNDEF;
 }
 
+void AMDGPUTargetLowering::replaceCTLZResults(
+    SDValue Op, SelectionDAG &DAG, SmallVectorImpl<SDValue> &Results) const {
+  auto SL = SDLoc(Op);
+  auto Arg = Op.getOperand(0u);
+  auto ResultVT = Op.getValueType();
+
+  if (ResultVT != MVT::i8)
```

Comment:
This isn't a problem really specific to i8; The same issue arises for i16 when it's not legal. Really should cover any sub-32-bit type

Reply:
You will see the same issue for i16 for targets where i16 is not legal. e.g. -mcpu=hawaii